ADAL/src/public/ADAuthenticationContext.h

```diff
@@ -451,6 +451,23 @@ typedef enum
                          redirectUri:(NSURL *)redirectUri
                      completionBlock:(ADAuthenticationCallback)completionBlock;
 
+/*! Follows the OAuth2 protocol (RFC 6749). The function will use the refresh token provided to get access token.
+ This method will not show UI for the user to reauthorize resource usage.
+ If the call fails, error will be included in the result.
+ @param refreshToken The refresh token provided in order to get acces token.
+ @param resource The resource whose token is needed.
+ @param clientId The client identifier
+ @param redirectUri The redirect URI according to OAuth2 protocol
+ @param userId An ADUserIdentifier object describing the user being authenticated
+ @param completionBlock The block to execute upon completion. You can use embedded block, e.g. "^(ADAuthenticationResult res){ <your logic here> }"
+ */
+- (void)acquireTokenWithRefreshToken:(NSString *)refreshToken
+                            resource:(NSString *)resource
+                            clientId:(NSString *)clientId
+                         redirectUri:(NSURL *)redirectUri
+                      userIdentifier:(ADUserIdentifier *)userId
```

Comment:
I don't think it makes sense to require ADUserInformation, because this is essentially a silent flow. In silent flows we normally ask for NSString userId.

Reply:
Agree, should we treat it as `OptionalDisplayableId` or `RequiredDisplayableId`?

Reply:
Let's use the same logic as normal acquire token silent.

Reply:
👌  our current default value is `RequiredDisplayableId`